tests/Api/Response/ResponseParserTest.php

```diff
@@ -79,6 +79,46 @@ public function testParse()
         ]);
     }
 
+    public function testHandleErrorResponse()
+    {
+        $errorResponse = [
+            'error' => true,
+            'error_no' => 0,
+            'error_msg' => 'Curl error: Could not resolve host: service.findologic.com',
+            'error_file' => '/findologic/http_request2/HTTP/Request2/Adapter/Curl.php',
+            'error_line' => 155
+        ];
+        $expectedExceptionMessage = 'simplexml_load_string() expects parameter 1 to be string, array given';
```

Comment:
We shouldn't even get to the point where `simplexml_load_string` tries to parse an array. We must catch this before and handle this case properly, without relying that passing data that can not be parsed by the parser, throwing an exception. Issues that we know beforehand that can be catched before an actual exception occurs should always be preferred.

For this please check if the data coming is an actual array and in case it is, log an appropriate error message e.g. `Response could not be parsed`, maybe even add the response as a log context.

Also since there is no exception, we do not need to log this.